crates/edr_eth/src/signature/y_parity.rs

```diff
@@ -1,80 +1,81 @@
 use alloy_rlp::BufMut;
 use k256::SecretKey;
+use revm_primitives::PrimitiveSignature;
 
 use super::{Recoverable, RecoveryMessage, Signature, SignatureError, SignatureWithRecoveryId};
 use crate::{Address, U256};
 
-#[derive(Clone, Copy, Debug, Eq, PartialEq, Hash)]
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
 #[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
 /// An ECDSA signature with Y-parity.
-pub struct SignatureWithYParity {
-    /// R value
-    pub r: U256,
-    /// S value
-    pub s: U256,
-    /// Whether the V value has odd Y parity.
-    pub y_parity: bool,
-}
+pub struct SignatureWithYParity(PrimitiveSignature);
 
 impl SignatureWithYParity {
+    /// Constructs a new instance from the provided `r`, `s`, and `y_parity`
+    /// values.
+    pub fn new(r: U256, s: U256, y_parity: bool) -> Self {
```

Comment:
Nit: it's easy to mix up `r` and `s`, because they're consecutive parameters with the same type. For me a struct providing named parameters would be better as argument, but it's not a blocker.

Reply:
Thanks for the suggestion. Suggestion applied [here](https://github.com/NomicFoundation/edr/pull/805/commits/ee256e26ea5e9bc73f9b6e8f30265ba47d62c057)